test/goaway_test.go

```diff
@@ -57,21 +57,20 @@ func (s) TestGracefulClientOnGoAway(t *testing.T) {
 	const maxConnAge = 100 * time.Millisecond
 	const testTime = maxConnAge * 10
 
+	lis, err := net.Listen("tcp", "localhost:0")
```

Comment:
Please consider using this instead: https://github.com/grpc/grpc-go/blob/master/internal/testutils/local_listener.go#L24

The following has a different implementation when tested inside of Google that works better with internal infrastructure.